### ISSUE TITLE
LiquidCore0.6.0 issue with iOS using CocoaPod

### ISSUE THREAD

**Issue by praveenkg85** [LiquidPlayer/LiquidCore#98]
@ericwlange , locally generated (through npm run bundler) liquid.bundle file works fine with Android. For iOS, it works fine only with Carthage but throws error with Cocoa Pod (invalid character '\uffff' at line...).
Although, it works fine with running the dev server, but due to some limitation, we have to run npm bundle for liquid core and generate liquid.bundle file.
Node:- "liquidcore-cli": "^0.4.7"
iOS:- liquidcore 0.6.0

Can you please help me resolve this error?

**Comment by ericwlange** [comment]
Interesting.  It works with the server but not the bundler?  That's very strange because the bundler simply runs an instance of the server and saves the output!

One thing you can try is to clone the repo and use the master version of LiquidCore (`path => /path/to/LiqudCore/LiquidCore.podspec` in your `Podfile`).  I made a change to the UTF8 library recently that will go into 0.6.1 that may help, but I am not sure.  Please try it and let me know.

**Comment by A-P-Singh** [comment]
Hi ,
I am trying to setup fresh new code for IOS (0.6.0) by using this [link](https://github.com/LiquidPlayer/LiquidCore/wiki/README-iOS-(0.6.0)#building-the-liquidcore-ios-framework) . 
I am getting error in pod lib lint .
Please find the attachment.
<img width="1680" alt="Screen Shot 2019-04-23 at 4 41 58 PM" src="https://user-images.githubusercontent.com/22491079/56576974-723d9500-65e7-11e9-9f6a-12568c51c105.png">
   plz suggest the solution.

**Comment by ericwlange** [comment]
Yes this was reported in issue #96. Please see workaround. The fix will be
in 0.6.1.

**Comment by umangkathuria** [comment]
Hi Eric!

I have been trying to find a solution to this issue for sometime. 

I have tried with xCode 10.1 and LiquidCore 0.6.0. I followed the steps as per the document - 
adding LiquidCore in my Podfile and performing a pod install. 

The moment I add my bundle and try to run the code- it throws me an error- "invalid character '\uffff' at line...)"

Also, I updated to xCode 10.2.1 with latest SDK and also implemented the workaround mentioned in issue #96, but ended up with the same issue. 

It would be really helpful if you could help me resolve this and/or share a working sample code in iOS Application that uses a bundle from the local(i.e. not from Server)

Also, when can we expect 0.6.1 to be released ?

Thank you!

**Comment by ericwlange** [comment]
Ok, I will check it out.  I guess I've never tried the bundle on iOS.

0.6.1 is pretty much ready.  I just need to go through the release process.  But it sounds like I should hold it until I've figured out this bug.  My goal is to get it out by end of week.  I am starting a new job next week, and will have less time to focus on this, so I wanted to get it out before then.

**Comment by ericwlange** [comment]
I have verified this happens on 0.6.0, but does _not_ happen on master.  So the small change I made to UTF8 handling seems to have fixed the problem.  Was anyone else able to verify this as well?

To test:

```
cd ~/ArbitraryDirectory
git clone https://github.com/LiquidPlayer/LiquidCore.git
```

In your project's `Podfile`, change:

```ruby
    pod 'LiquidCore', '0.6.0'
```

to

```ruby
   pod 'LiquidCore', 'path => ~/ArbitraryDirectory/LiquidCore/LiquidCore.podspec'
```

And then:

```
pod install
```

You will have the latest.  You should be able to use local bundles.

If that is verified, then I am ready to close 0.6.1 and push it out by the weekend.

**Comment by umangkathuria** [comment]
Thanks, Eric!
For future references: The issue seems to be resolved now. This is what I added in my podfile - 
pod 'LiquidCore', :git => 'https://github.com/LiquidPlayer/LiquidCore.git'

the one mentioned above was giving an error. Anyway, I can verify that the "Invalid character issue" with bundle is fixed.

Thank you so much for your time, effort and quick responses.

**Comment by alexanderzhigulich** [comment]
I confirm also,

Thank you very much!

**Comment by praveenkg85** [comment]
Thanks Eric for your continuous support and quick fix. I hope, we'll get this fix in 0.6.1 release.

**Comment by ericwlange** [comment]
@umangkathuria - Thanks for the tip!  I like CocoaPods more and more each day.  So much better than Carthage.

0.6.1 has been released with this fix in it.  Thanks all for the verification.
